entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/mapper/ConvertOperationBuilderImpl.java

```diff
@@ -83,6 +87,11 @@ public T convert() {
 
     @Override
     public ConvertOperationBuilder<T> convertAttribute(String attributePath, Class<?> attributeViewClass, String constructorName, ConvertOption... convertOptions) {
+        if (key.getSourceType() == null) {
+            subMappers.put(attributePath, (ViewMapper.Key<Object, Object>) ViewMapper.Key.create(entityViewManager.getMetamodel(), null, attributeViewClass, constructorName, convertOptions));
+            return this;
+        }
```

Comment:
We should actually also check if the target type is compatible with the given `attributeViewClass`.

```suggestion
        AbstractMethodAttribute<? super Object, ?> targetAttribute = key.getTargetType().getRecursiveSubviewAttributes().get(attributePath);
        if (targetAttribute == null) {
            throw new IllegalArgumentException("Attribute '" + attributePath + "' could not be found on type: " + key.getTargetType().getJavaType().getName());
        }
        Class<?> targetType = targetAttribute.getConvertedJavaType();
        if (targetAttribute instanceof PluralAttribute<?, ?, ?>) {
            targetType = ((PluralAttribute<?, ?, ?>) targetAttribute).getElementType().getJavaType();
        }
        if (!targetType.isAssignableFrom(attributeViewClass)) {
            throw new IllegalArgumentException("The given type '" + attributeViewClass.getName() + "' is not assignable to the declared type of '" + key.getTargetType().getJavaType().getName() + "." + attributePath + "': " + targetType.getName());
        }
        if (key.getSourceType() == null) {
            subMappers.put(attributePath, (ViewMapper.Key<Object, Object>) ViewMapper.Key.create(entityViewManager.getMetamodel(), null, attributeViewClass, constructorName, convertOptions));
            return this;
        }
```